main.js

```diff
@@ -1,30 +1,45 @@
-const fs = require('fs');
-
-const TASKS_FILE = 'tasks.json'; // Файл, де зберігаються завдання
-
-function saveTasks(tasks) {
-  fs.writeFileSync(TASKS_FILE, JSON.stringify(tasks, null, 2));
-}
-
-function showIncompleteTasks() {
-  const tasks = loadTasks();
-  const incompleteTasks = tasks.filter((task) => !task.completed);
-  incompleteTasks.sort((a, b) => new Date(a.deadline) - new Date(b.deadline));
-  console.log('Incomplete Tasks:');
-  incompleteTasks.forEach((task) => {
-    console.log(`- ${task.title} (Deadline: ${task.deadline})`);
-  });
-}
-
-function addTask(title, description, deadline) {
-  const tasks = loadTasks();
-  const task = {
-    title,
-    description,
-    deadline,
-    completed: false,
-  };
-  tasks.push(task);
-  saveTasks(tasks);
-  console.log(`Task "${title}" added.`);
-}
+const {
+    getIncompleteTasks,
+    getAllTasks,
+    markTaskAsCompleted,
+    addTask,
+    editTask,
+    getOverdueTasks,
+    deleteTask
+  } = require('./taskManager');
+  
+  if (process.argv.length < 3) {
+    console.log('Usage: node app.js <command> [arguments]');
+    process.exit(1);
+  }
+  
+  const command = process.argv[2];
+  
+  if (command === 'get-incomplete') {
+    getIncompleteTasks();
+  } else if (command === 'get-all') {
+    getAllTasks();
+  } else if (command === 'mark-completed') {
+    const taskTitle = process.argv[3];
+    markTaskAsCompleted(taskTitle);
+  } else if (command === 'add') {
+    const title = process.argv[3];
+    const description = process.argv[4];
+    const deadline = process.argv[5];
+    addTask(title, description, deadline);
+  } else if (command === 'edit') {
+    const taskTitle = process.argv[3];
+    const newTitle = process.argv[4];
+    const newDescription = process.argv[5];
+    const newDeadline = process.argv[6];
+    editTask(taskTitle, newTitle, newDescription, newDeadline);
+  } else if (command === 'get-overdue') {
+    getOverdueTasks();
+  } else if (command === 'delete') {
+    const taskTitle = process.argv[3];
+    deleteTask(taskTitle);
+  } else {
+    console.log('Invalid command.');
+    process.exit(1);
+  }
```

Comment:
Refactor this is-else into map collection, like this

```suggestion
  const commandActions = {
  'get-incomplete': getIncompleteTasks,
  'get-all': getAllTasks,
  'mark-completed': () => {
    const taskTitle = process.argv[3];
    markTaskAsCompleted(taskTitle);
  },
  'add': () => {
    const title = process.argv[3];
    const description = process.argv[4];
    const deadline = process.argv[5];
    addTask(title, description, deadline);
  },
  'edit': () => {
    const taskTitle = process.argv[3];
    const newTitle = process.argv[4];
    const newDescription = process.argv[5];
    const newDeadline = process.argv[6];
    editTask(taskTitle, newTitle, newDescription, newDeadline);
  },
  'get-overdue': getOverdueTasks,
  'delete': () => {
    const taskTitle = process.argv[3];
    deleteTask(taskTitle);
  }
};

if (command in commandActions) {
  commandActions[command]();
} else {
  console.log('Invalid command.');
  process.exit(1);
}
```